app/Http/Controllers/RuleEngine/RuleExecutionController.php

```
@@ -0,0 +1,405 @@
+                'total_rules' => count($ruleIds),
+                'total_transactions' => $transactions->count(),
+                'total_rules_matched' => count(array_filter($results, fn ($r) => ! empty($r['actions']))),
+                'results' => $results,
+            ],
+        ]);
+    }
+
+    /**
+     * Get descriptions of what actions would do.
+     */
+    private function getActionsDescription($actions): array
```

PHPStan (2.1.15)

Method App\Http\Controllers\RuleEngine\RuleExecutionController::getActionsDescription() has parameter $actions with no type specified.


(missingType.parameter)